### PR TITLE
Add detailed channel info

### PR DESCRIPTION
**Pull request by burnsed**
This includes https://github.com/Clever/rabbit-hole/pull/4 and as mentioned there, this change does not account for `unknown` consumers (nor does the rest of rabbit-hole). Golang makes handling this case pretty involved and I think it is sufficient for the client of this code to simply check if the error returned is from unmarshaling and if so, work around it there. Also, we seem to be one of the very few users who experience `unknown` consumers.

### CODE REVIEW

**Comment by burnsed** [comment]
cc @nathanleiby: I want to make sure you are cool with this change

File: exchanges.go [hunk 1]
```diff
@@ -126,11 +126,11 @@ func (c *Client) ListExchangesIn(vhost string) (rec []ExchangeInfo, err error) {
...
 
 type ExchangeIngressDetails struct {
-	Stats          MessageStats      `json:"stats"`
-	ChannelDetails PublishingChannel `json:"channel_details"`
+	Stats          MessageStats        `json:"stats"`
+	ChannelDetails BriefChannelDetails `json:"channel_details"`
 }
 
-type PublishingChannel struct {
+type BriefChannelDetails struct {
```
Comment (nathanleiby) [inline exchanges.go]: Not sure I see why s/PublishingChannel/BriefChannelDetails ... as far as I can tell it's just renaming it?

File: channels.go [hunk 2]
```diff
@@ -12,6 +12,16 @@ type BriefConnectionDetails struct {
...
 }
 
+type BriefConsumerDetails struct {
+	Arguments      map[string]interface{} `json:"arguments"`
+	PrefetchCount  int                    `json:"prefetch_count"`
+	AckRequired    bool                   `json:"ack_required"`
+	Exclusive      bool                   `json:"exclusive"`
+	ConsumerTag    string                 `json:"consumer_tag"`
+	Queue          NameAndVhost           `json:"queue"`
+	ChannelDetails BriefChannelDetails    `json:"channel_details"`
```
Comment (nathanleiby) [inline channels.go]: nit: these we alphabetically sorted in https://github.com/Clever/rabbit-hole/pull/4 , which made them a little easier to parse

File: channels.go [hunk 3]
```diff
@@ -12,6 +12,16 @@ type BriefConnectionDetails struct {
 	PeerHost string `json:"peer_host"`
 }
 
+type BriefConsumerDetails struct {
```
Comment (nathanleiby) [inline channels.go]: nit: comment to name this and either mention (1) exactly what's missing or (2) text like "(very incomplete)"

**Comment by nathanleiby** [comment]
minor comments, lgtm overall too

File: channels.go (hunk 3 above)
Comment (burnsed) [inline channels.go]: This technically isn't missing anything from the response that is returned. Some APIs like `list*`, just return a subset of consumer information and then to get the full consumer info, you would GET on the consumer in question.

File: exchanges.go (hunk 1 above)
Comment (burnsed) [inline exchanges.go]: Yes, this is just a rename to make it consistent with the naming convention used in other similar cases.

File: channels.go (hunk 2 above)
Comment (burnsed) [inline channels.go]: good idea. will do...

**Pull request merged**